test/e2e/snaps/test-snap-installed-metric.spec.js

```diff
@@ -0,0 +1,107 @@
+const { strict: assert } = require('assert');
+const {
+  defaultGanacheOptions,
+  withFixtures,
+  unlockWallet,
+  getEventPayloads,
+} = require('../helpers');
+const FixtureBuilder = require('../fixture-builder');
+const { TEST_SNAPS_WEBSITE_URL } = require('./enums');
+
+async function mockSegment(mockServer) {
+  return [
+    await mockServer
+      .forPost('https://api.segment.io/v1/batch')
+      .withJsonBodyIncluding({
+        batch: [{ type: 'track', event: 'Snap Installed' }],
+      })
+      .thenCallback(() => {
+        return {
+          statusCode: 200,
+        };
+      }),
+  ];
+}
+
+describe('Snap Installed Event', function () {
+  it('successfully tracked when snap is installed', async function () {
+    await withFixtures(
+      {
+        fixtures: new FixtureBuilder()
+          .withMetaMetricsController({
+            metaMetricsId: 'fake-metrics-id',
+            participateInMetaMetrics: true,
+          })
+          .build(),
+        defaultGanacheOptions,
+        title: this.test.title,
+        testSpecificMock: mockSegment,
+      },
+      async ({ driver, mockedEndpoint: mockedEndpoints }) => {
+        await driver.navigate();
+
+        await unlockWallet(driver);
+
+        // navigate to test snaps page and connect
+        await driver.openNewPage(TEST_SNAPS_WEBSITE_URL);
+        await driver.delay(1000);
+        const confirmButton = await driver.findElement('#connectDialogSnap');
+        await driver.scrollToElement(confirmButton);
+        await driver.delay(500);
+        await driver.clickElement('#connectDialogSnap');
+        await driver.delay(500);
+
+        // switch to metamask extension and click connect
+        const windowHandles = await driver.waitUntilXWindowHandles(
+          3,
+          1000,
+          10000,
+        );
+
+        await driver.switchToWindowWithTitle(
+          'MetaMask Notification',
+          windowHandles,
+        );
+
+        await driver.clickElement({
+          text: 'Connect',
+          tag: 'button',
+        });
+
+        await driver.waitForSelector({ text: 'Install' });
+
+        await driver.clickElement({
+          text: 'Install',
+          tag: 'button',
+        });
+
+        await driver.waitForSelector({ text: 'OK' });
```

Comment:
I don't believe we need to use `await driver.waitForSelector({ text: *** });` before each clickElement. Under the hood, clickElement is calling findElement, which waits until the 10 s timeout to find the element.

Reply:
There seems to be a StaleElementReferenceError thrown, likely the result of the 2 `await driver.clickElement('[data-testid="page-container-footer-next"]');`. It might be worth exploring waiting for the second screen to load in between clicks.

Reply:
Ideally, our data-testid values should be unique, but that's out of scope in relation to this PR